File: src/main/java/io/junseok/mealmate/exception/GlobalExceptionHandler.java
package io.junseok.mealmate.exception;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
public class GlobalExceptionHandler {
    @ExceptionHandler(MealMateException.class)
    protected ResponseEntity<ErrorResponseEntity> errorCodeResponseEntity(MealMateException ex){
        return ErrorResponseEntity.responseEntity(ex.getErrorCode());
    }
}

File: src/main/java/io/junseok/mealmate/exception/GlobalExceptionHandler.kt
package io.junseok.mealmate.exception

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class GlobalExceptionHandler {
    @ExceptionHandler(MealMateException::class)
    protected fun errorCodeResponseEntity(ex: MealMateException): ResponseEntity<ErrorResponseEntity> =
        ex.errorCode.responseEntity()
}
